pkg/virt-launcher/virtwrap/api/converter.go

```diff
@@ -508,11 +508,30 @@ func Convert_v1_FeatureHyperv_To_api_FeatureHyperv(source *v1.FeatureHyperv, hyp
 	return nil
 }
 
-func popSRIOVPCIAddress(addrs []string) (string, []string, error) {
-	if len(addrs) > 0 {
-		return addrs[0], addrs[1:], nil
+func filterAddress(addrs []string, addr string) []string {
+	var res []string
+	for _, a := range addrs {
+		if a != addr {
+			res = append(res, a)
+		}
 	}
-	return "", addrs, fmt.Errorf("no more SR-IOV PCI addresses to allocate")
+	return res
+}
+
+func reserveAddress(addrsMap map[string][]string, addr string) {
+	for networkName, addrs := range addrsMap {
+		addrsMap[networkName] = filterAddress(addrs, addr)
+	}
+	return
+}
+
+func popSRIOVPCIAddress(networkName string, addrsMap map[string][]string) (string, map[string][]string, error) {
```

Comment:
Could you please add a docstring here? I don't really get what and why are we doing here.

Reply:
@phoracek my understanding is that we are poping addresses that have been assigned to devices on a certain network.